### ISSUE TITLE
How to define my own loss function in Keras?

### ISSUE THREAD

**Issue by zhjmcjk** [keras-team/keras#10396]
Hi, folks. I am trying using CuDNNLSTM in Keras to deal with some sequential questions with multiple outputs. My code is written as follows:

```
inputs = Input(...)
lstm_outputs = CuDNNLSTM(n_neurons, return_sequences=True)(inputs)
outputs = TimeDistributed(Dense(3))(lstm_outputs)
model = Model(inputs=inputs, outputs=outputs)
```

When I using the mean_square_error as loss function in Keras, the results seem like
error = np.mean(np.square(y_pred - y_real)), where the feature number of y_pred and y_real is 3 in my case.

But I want to define the loss function as following:
error = np.mean(np.square(y_pred[:, :, 0] - y_real[:, :, 0]) + np.square(y_pred[:, :, 1] - y_real[:, :, 1]) + np.square(y_pred[:, :, 2] - y_real[:, :, 2]))

I don't know how to implement this, can anyone give me some suggestions? Thanks a lot!

**Comment by shivam05011996** [comment]
You need to import keras backend, i will demonstrate your loss function, please modify according to your slices
```
from keras import backend as K
>>> def loss(y_true, y_pred):
...     return K.mean(K.square(y_pred - y_true) + K.square(y_pred - y_true) + K.square(y_pred - y_true))
... 
>>> loss(4,5)
<tf.Tensor 'Mean:0' shape=() dtype=int32>
```

**Comment by zhjmcjk** [comment]
Wow, excellent, thanks for your reply. But one question: if the features number of y_real and y_pred is 3, like y_real=[1; 2; 3], y_predict=[1.2; 2.1; 3.5], then the loss function can be written by following:

```
from keras import backend as K
def my_loss_function(y_real, y_pred):
      return K.mean(K.square(y_pred[:, :, 0] - y_true[:, :, 0]) + K.square(y_pred[:, :, 1] - y_true[:, :, 1]) + 
                      K.square(y_pred[:, :, 2] - y_true[:, :, 2]), axis=-1)

```
Am I right? @shivam05011996

**Comment by shivam05011996** [comment]
@zhjmcjk I believe so, but please verify this, instead of taking a long shot to keras loss, implement the loss function in numpy, then see if the result outputted meets your expectations or not. like - 
return np.mean(np.square(......))

**Comment by lilaczhou** [comment]
@zhjmcjk @shivam05011996 
I want to implement a loss function like this
![image](https://user-images.githubusercontent.com/45725450/59413944-2b079f80-8df3-11e9-8ec7-c77863761f5b.png)
it means while y_true>0,loss=y_true*log(y_pred);else loss=y_true*log(1-y_pred)
I try to code like this 
![image](https://user-images.githubusercontent.com/45725450/59414189-a5382400-8df3-11e9-9c63-24ea0a8af3a7.png)
but error occurs
![image](https://user-images.githubusercontent.com/45725450/59414229-b5e89a00-8df3-11e9-85fb-4d5620f27209.png)

I can't solve the problem,could you please help me ?
